packages/bpk-component-phone-input/src/BpkPhoneInput.js

```diff
@@ -111,28 +111,30 @@ const BpkPhoneInput = (props: Props) => {
   let phoneDisplayValue;
 
   if (countryCodeMask) {
-    if (value.startsWith('+')) {
-      if (value.startsWith(`+${dialingCodeText} `)) {
-        phoneDisplayValue = value;
-      } else if (value.charAt(`+${dialingCodeText}`.length) !== ' ') {
-        const phoneNumber = value.slice(`+${dialingCodeText}`.length);
-        phoneDisplayValue = `+${dialingCodeText} ${phoneNumber}`;
-      } else {
-        const phoneValue = value.split(' ')[1];
-
-        phoneDisplayValue = phoneValue
-          ? `+${dialingCodeText} ${phoneValue}`
-          : `+${dialingCodeText} `;
-      }
-    } else if (value.includes(dialingCodeText)) {
-      phoneDisplayValue = `+${value}`;
-    } else {
-      phoneDisplayValue = `+${dialingCodeText} ${value}`;
-    }
+    phoneDisplayValue = `+${dialingCodeText} ${value}`;
   } else {
     phoneDisplayValue = value;
   }
 
+  const phoneNumberOnChange = formFieldEvt => {
+    let { value: phoneValueWithCode } = formFieldEvt.target;
+    const { name: fieldPath } = formFieldEvt.target;
+
+    if (phoneValueWithCode.charAt(`+${dialingCodeText}`.length) !== ' ') {
+      const number = phoneValueWithCode.slice(`+${dialingCodeText}`.length);
+      phoneValueWithCode = `+${dialingCodeText} ${number}`;
+    }
+
+    const phoneNumber = phoneValueWithCode.split(' ')[1] || '';
+    const passThroughEvent = {
```

Comment:
I'm just afraid we won't be able to merge it back to master like this. This is removing all other properties from the event, like `preventDefault` that people might be using. What do you think of just mutating the event object and returning the original event instead?

Reply:
great shout - feedback applied

Reply:
this will still result in an error @Andi24. 

```
const a = new Event('MouseEvent')
a.preventDefault()
// undefined
const b = { ...a }
b.preventDefault()
// TypeError: preventDefault is not a function
```

Spreading the object will only copy the properties not apply the prototype, I think the simplest solution is to just mutate the original event.


```formFieldEvt.target.value = newValue```

Reply:
the issue with mutating the original event object is getting an ESlint error - do you think it's more desirable to make an eslint exception block on this param reassing than losing the prototype?

Reply:
This is a case where disabling the eslint rule is acceptable, it's better than losing the event. Another solution would be to call `onChange` with two arguments.

`onChange(originalEvent, phoneNumber)`. This will be a breaking change but it seems to me this is the best solution.

What do you think?

Reply:
Refactoring the `onChange` to take parameters would mean adding an exception to read the `value` from the parameter rather than the received value in a method that is generically used across multiple fields in the context where the phone input is used. I would rather add an eslint exception in this case.

Reply:
disabled eslint rule